src/main/java/io/pinecone/clients/Pinecone.java

```diff
@@ -1034,13 +1046,23 @@ public Builder withProxy(String proxyHost, int proxyPort) {
          */
         public Pinecone build() {
             PineconeConfig config = new PineconeConfig(apiKey, sourceTag, proxyConfig, customOkHttpClient);
+            config.setTLSEnabled(enableTls);
             config.validate();
 
             if (proxyConfig != null && customOkHttpClient != null) {
                 throw new PineconeConfigurationException("Invalid configuration: Both Custom OkHttpClient and Proxy are set. Please configure only one of these options.");
             }
 
-            ApiClient apiClient = (customOkHttpClient != null) ? new ApiClient(customOkHttpClient) : new ApiClient(buildOkHttpClient(proxyConfig));
+            ApiClient apiClient;
+            if (customOkHttpClient != null) {
+                apiClient = new ApiClient(customOkHttpClient);
+            } else {
+                apiClient = new ApiClient(buildOkHttpClient(proxyConfig));
+                if(host!=null && !host.isEmpty()) {
+                    config.setHost(host);
+                    apiClient.setBasePath(host);
```

Comment:
If the host is applied to the `apiClient` does it also need to be passed to `config` here? From what it looks like, this value is overwritten when you target an index: https://github.com/pinecone-io/pinecone-java-client/blob/main/src/main/java/io/pinecone/clients/Pinecone.java#L844-L874

Reply:
Currently we are making an extra describeIndex call using the same Pinecone object so the overwritten value is still pointing to the same host but this extra call can be removed. The only approach I came up with for now, in order to solve this, is instantiating the PineconeConfig object outside of Builder class which I don't like it. So I have a ticket to remove this extra call in future and come up with a better solution. Also the ticket will take care of asyncIndexConnection as well.